pkg/cloudmap/client.go

```diff
@@ -147,36 +145,27 @@ func (sdc *serviceDiscoveryClient) RegisterEndpoints(ctx context.Context, nsName
 
 	sdc.log.Info("registering endpoints", "namespaceName", nsName, "serviceName", svcName, "endpoints", endpts)
 
-	svcIdMap, err := sdc.getServiceIds(ctx, nsName)
+	svcId, err := sdc.getServiceId(ctx, nsName, svcName)
 	if err != nil {
 		return err
 	}
-	svcId, found := svcIdMap[svcName]
-	if !found {
-		return fmt.Errorf("service not found in Cloud Map: %s", svcName)
-	}
-
-	opCollector := NewOperationCollector()
 
+	operationPoller := NewOperationPoller(sdc.sdApi)
 	for _, endpt := range endpts {
-		endptId := endpt.Id
 		endptAttrs := endpt.GetCloudMapAttributes()
-		opCollector.Add(func() (opId string, err error) {
-			return sdc.sdApi.RegisterInstance(ctx, svcId, endptId, endptAttrs)
-		})
+		opId, endptErr := sdc.sdApi.RegisterInstance(ctx, svcId, endpt.Id, endptAttrs)
```

Comment:
This will make registerInstance calls one at a time, the collector was designed used channels to submit multiple in parallel.

Reply:
Fair point. Resolved.